app/validation/validator.py

```diff
@@ -1,6 +1,98 @@
+from app.validation.validation_result import ValidationResult
+from app.validation.required import Required
+from app.validation.type_validator_factory import TypeValidatorFactory
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class ValidationException(Exception):
+    pass
+
+
 class Validator(object):
-    def set_context(self, context):
-        raise NotImplementedError()
+    def __init__(self, schema, validation_store, response_store):
+        self._schema = schema
+        self._validation_store = validation_store
+        self._response_store = response_store
+
+        # Set the factory class here, so we can override it for tests
+        self._type_validator_factory_class = TypeValidatorFactory
+
+    def validate(self, user_data):
+        for item_id in user_data.keys():
+            logger.debug('Attempting to validate {}'.format(item_id))
+            item = self._schema.get_item_by_id(item_id)
+
+            # Currently the datefields trip this up
+            if not item:
+                logger.info('Ignoring {}'.format(item_id))
```

Comment:
I've left it at info for now as it highlights the issue we have around date handling.  Ultimately, we shouldn't be ignoring anything without good reason